warehouse/query-core/src/main/java/datawave/query/util/AllFieldMetadataHelper.java

```diff
@@ -507,7 +507,7 @@ public CompositeMetadata getCompositeMetadata(Set<String> ingestTypeFilter) thro
     @Cacheable(value = "getFieldsForDatatype", key = "{#datawaveType,#ingestTypeFilter}", cacheManager = "metadataHelperCacheManager")
```

Comment:
I think you could add `#root.target.metadataTableName` to the key to cover this, unless metadataTableName needs to be pushed into the method signature...

Reply:
You didn't change it in this PR, but there's also a caching problem on the method `public Set<String> getFieldsForDatatype(Class<? extends Type<?>> datawaveType)`. Internally, it calls this method. However, since it's an internal call, the caching annotation on this method won't be honored. Therefore, if you modify the cache key on this method to include the metadataTableName variable, you must also modify the cache key on the ``public Set<String> getFieldsForDatatype(Class<? extends Type<?>> datawaveType)` variant.

Reply:
I don't believe that the metadataTableName will ever change in the MetadataHelper or in the AllFieldMetadataHelper, once the instance is initialized.